### PR TITLE
06986 Update platform status enum

### LINKED ISSUE #6986: Update platform status enum
Update the platform status enum to prepare for implementation of #6297, as proposed in #6296

### PR DESCRIPTION
**Pull request by alittley**
Closes #6986 

Merging of this PR must wait until the changes the changes tracked in https://github.com/swirlds/swirlds-platform-regression/issues/3484 have been made and are ready to merge

DEVOPS IMPACT: This change will affect DevOps automation and alerting around detecting freeze. DevOps must be notified of the services release this change will be contained in, to account for the change and plan for any necessary updates

### CODE REVIEW

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import com.swirlds.common.UniqueId;
+
+/**
+ * The status of the Platform
+ * <p>
+ * NOTE: not all of these statuses can currently be reached. The status state machine is still under development. In
+ * such cases, the documentation will indicate that the status is "not in use"
+ *
+ */
+public enum PlatformStatus implements UniqueId {
```
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java:28]: nitpick, some javadocs end with a `.`, but others do not. My personal preference in javadocs is to end sentences with `.` if writing full sentences.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    STARTING_UP(1),
+    /**
+     * The platform is gossiping, creating events, and accepting app transactions
+     */
+    ACTIVE(2),
+    /**
+     * The platform is not currently connected to any other computers on the network
+     */
+    DISCONNECTED(3),
```
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java:40]: Add a note that this status is scheduled to be retired

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+     */
+    FREEZE_COMPLETE(6),
+    /**
+     * The platform is replaying events from the preconsensus event stream
+     * <p>
+     * NOTE: not in use
+     */
+    REPLAYING_EVENTS(7),
+    /**
+     * The platform has just started, and is observing the network. The platform is gossiping, but will not produce
```
Comment (cody-littley) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatus.java:61]:
```suggestion
     * The platform has just started, and is observing the network. The platform is gossiping, but will not create
```

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by cody-littley — APPROVED** [review]

File: platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java [hunk 4]
```diff
@@ -841,7 +841,7 @@ private void platformStatusChange(final PlatformStatusChangeNotification notific
         // set isActive
         isActive = newStatus == PlatformStatus.ACTIVE;
 
-        if (newStatus == PlatformStatus.MAINTENANCE) {
+        if (newStatus == PlatformStatus.FREEZING) {
             logger.trace(LOGM_DEMO_INFO, "ENTERING MAINTENANCE!");
```
Comment (poulok) [inline platform-sdk/platform-apps/tests/PlatformTestingTool/src/main/java/com/swirlds/demo/platform/PlatformTestingToolMain.java:845]: Please also update the logs for consistency.

**Review by lpetrovic05 — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by hendrikebbers — DISMISSED** [review]

**Review by cody-littley — APPROVED** [review]

**Pull request merged**